webapp/packages/core-blocks/src/ResourcesHooks/useResource.ts

```diff
@@ -161,7 +161,12 @@ export function useResource<
       propertiesRef.includes = includes;
     }
 
-    if (key === null || propertiesRef.key === null || !propertiesRef.resource.isIntersect(key, propertiesRef.key)) {
+    if (
+      key === null ||
+      propertiesRef.key === null ||
+      !propertiesRef.resource.isIntersect(key, propertiesRef.key) ||
+      !propertiesRef.resource.isEqual(key, propertiesRef.key)
```

Comment:
May be we should put a `shouldCompareEqual` flag in useResource arguments
and use it here like `!propertiesRef.resource.isEqual(key, propertiesRef.key)` if we dont want extra calculations for equality

Reply:
Seems like isIntersect is not suitable for useResource anymore because our data depends on the key, and in some cases, we want this data not just to include data that we need but also we don't want data to contain extra data